### PR TITLE
Use evalsha for scripts

### PR DESCRIPTION
**Pull request by yosiat**
Lazy loading the scripts: trying with evalsha if it fails on “NO SCRIPT” we will run eval (which will load the script) and next call will use evalsha.

https://github.com/mike-marcacci/node-redlock/issues/76

### CODE REVIEW

**Comment by yosiat** [comment]
@mike-marcacci can you please look at this PR?

File: redlock.js [hunk 1]
```diff
@@ -429,4 +419,24 @@ Redlock.prototype._random = function _random(){
 	return crypto.randomBytes(16).toString('hex');
 };
 
+Redlock.prototype._executeScript = function(server, name, args, callback) {
+  const script = this.scripts[name];
+
+  return server.evalsha(script.hash, args, (err, result) => {
+    if(err !== null && err.message.startsWith("NOSCRIPT")) {
+      // load the script, but don't wait for it, on the next call it will be available
+      server.script('load', script.value);
```
Comment (ramhr) [inline redlock.js:428]: If you call regular `eval` later you don't need to call `load`, as it automatically caches the script.
https://redis.io/commands/eval#script-cache-semantics
> Executed scripts are guaranteed to be in the script cache of a given execution of a Redis instance forever. This means that if an EVAL is performed against a Redis instance all the subsequent EVALSHA calls will succeed.

Comment (yosiat) [reply redlock.js]: thanks, fixed.

**Comment by yosiat** [comment]
@mike-marcacci can you please CR this PR?

**Comment by yosiat** [comment]
@mike-marcacci if you can, please look at this PR :)
It can reduce bandwidth to redis a lot...

**Review by mike-marcacci — APPROVED** [review]

**Comment by mike-marcacci** [comment]
Sorry for the delay - it's been a busy year 😬

Hoping to cut a new release shortly.

**Comment by yosiat** [comment]
@mike-marcacci thanks a lot, totally understandable :)

**Pull request merged**